### PR TITLE
Fix/rehydration

### LINKED ISSUE #1316: Orleans based Aggregate Roots does not need to rehydrate from the beginning if the grain is already in memory
We should have a similar mechanism to what we have for immediate projections were it doesn't rehydrate all events relevant if the grain is already in memory. Instead it should just apply any new events since the last rehydration.

### PR DESCRIPTION
**Pull request by woksin**
Fixed #1316

### CODE REVIEW

**Comment by woksin** [comment]
This is not completely done yet. I would like to have more integration tests

File: Source/Clients/AspNetCore/Transactions/UnitOfWorkMiddleware.cs [hunk 1]
```diff
@@ -21,6 +21,7 @@ public class UnitOfWorkMiddleware(IUnitOfWorkManager unitOfWorkManager, RequestD
     /// <returns>Awaitable task.</returns>
     public async Task InvokeAsync(HttpContext context)
     {
+        // TODO: Create Issue: Maybe there should be a try catch here and dispose the unit of work if an exception is thrown.
```
Comment (einari) [inline Source/Clients/AspNetCore/Transactions/UnitOfWorkMiddleware.cs:24]: What could cause an exception here? Are you seeing exceptions?
From what I can see from the current implementation all this does is create a new UnitOfWork instance and return this.

File: Source/Clients/DotNET/Aggregates/AggregateRootContext.cs [hunk 2]
```diff
@@ -33,5 +35,5 @@ public class AggregateRootContext(
     public IUnitOfWork UnitOfWOrk { get; } = unitOfWork;
 
     /// <inheritdoc/>
-    public bool HasEventsForRehydration { get; set; }
+    public EventSequenceNumber NextSequenceNumber { get; set; } = nextSequenceNumber;
```
Comment (einari) [inline Source/Clients/DotNET/Aggregates/AggregateRootContext.cs:38]: We can probably remove the setter here.

**Review by einari — CHANGES_REQUESTED** [review]
Great work! A few minor things to address.

File: Source/Clients/DotNET/Aggregates/IAggregateRootContext.cs [hunk 3]
```diff
@@ -35,7 +35,7 @@ public interface IAggregateRootContext
     IUnitOfWork UnitOfWOrk { get; }
 
     /// <summary>
-    /// Gets the last processed <see cref="EventSequenceNumber"/>.
+    /// Gets the next <see cref="EventSequenceNumber"/> to process for the aggregate root in the unit of work.
```
Comment (einari) [inline Source/Clients/DotNET/Aggregates/IAggregateRootContext.cs:38]:
```suggestion
    /// Gets or sets the next <see cref="EventSequenceNumber"/> to process for the aggregate root in the unit of work.
```

**Review by einari — CHANGES_REQUESTED** [review]

**Pull request merged**